### ISSUE TITLE
AutoSuggestBox control Accessibility - When a user types into the search box, narrator does not announce the number of suggestions shown

### ISSUE THREAD

**Issue by YuliKl** [microsoft/microsoft-ui-xaml#2043]
**Describe the bug**

The documentation suggests that the AutoSuggestBox control will provide the following Accessibility inbox, but the number of suggestions is never read and "Suggestions Available" is only sometimes read. 

>**Accessibility**
If you are using an assistive technology, such as Narrator, to interact with the AutoSuggestBox the accessibility experience has already been hooked up for you. A user will:
>- Know the list is present and when the list closes
>- Know how many suggestions are available
>- Be able to move Narrator focus to the list
>- Be able to Navigate through a suggestion with all other reading modes See Auto-suggest accessibility for more information.

**Additional context**

Copied from [internal bug](https://microsoft.visualstudio.com/OS/_workitems/edit/25277622/)

**Comment by ranjeshj** [comment]
This looks like a documentation issue. Please check if docs need to be updated or this is something that we need to fix in the control.

**Comment by michael-hawker** [comment]
I can confirm this behavior. We discovered this when we copied the template for the `RichSuggestBox` and noticed the same behavior. "Suggestions/Suggestions Available" is only sometimes read (but you don't know how many), I think it may depend on if you wait for the initial description to be read in full before typing the first time or if you've already performed a query and the suggestion list is still open.

When the pop-up of suggestions appear, nothing is read in narrator, so someone wouldn't know a list of search results has appeared. It should announce that there are search results available.

In the sample app, it's a bit weird too as the first example isn't keyboard accessible, filed a bug here for that: https://github.com/microsoft/Xaml-Controls-Gallery/issues/759

I imagine using `LandmarkType` and `LiveSetting` to the pop-up may help here in the existing template, so not sure if WinUI 3 is required here to at least investigate if there's a possible solution with the existing accessibility XAML helpers?

Repro steps:
1. Open the XAML Controls Gallery
2. Open the AutoSuggestBox example page
3. Turn on Narrator
4. Click in the AutoSuggestBox in the 2nd example "An AutoSuggestBox that provides a Searchbox experience"
5. Type 'a'
6. Note, pop-up of results appear, but Narrator is sometimes silent (type a fast and it's usually just saying "'a' edit"). Expect narrator to announce a search result list of X search results consistently. Note though that retyping in the box after initialization usually works.

**Comment by michael-hawker** [comment]
Sounds like this is still an issue, bumping as per #8638
